components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -764,7 +764,11 @@ export abstract class AbstractTypeORMWorkspaceDBImpl implements WorkspaceDB {
 
     public async findVolumeSnapshotById(volumeSnapshotId: string): Promise<VolumeSnapshot | undefined> {
         const volumeSnapshots = await this.getVolumeSnapshotRepo();
-        return volumeSnapshots.findOne(volumeSnapshotId);
+        return volumeSnapshots
+            .createQueryBuilder("vs")
+            .where("vs.deleted = 0")
+            .andWhereInIds(volumeSnapshotId)
```

Comment:
What's the difference between `findOne` and `andWhereInIds`?

Reply:
Wondering the same. Using `.andWhere("vs.id = :id", { id: volumneSnapshotId })` would be more direct, and require less TypeORM knowledge.

Reply:
Thanks, I've rebased to make this change.